tests/cpgrid/lgr/LgrChecks.hpp

```diff
@@ -817,5 +825,32 @@ void Opm::checkGlobalActiveCellsCountInGridWithLgrs(const Dune::CpGrid& grid,
 }
 
 
+void Opm::checkMarksAfterPreAdapt(const Dune::CpGrid& grid,
+                                  bool preAdapt)
+{
+    if (preAdapt) { // at least one element has been marked
+        for (const auto& element : Dune::elements(grid.leafGridView())) {
+            if (grid.getMark(element) == 1) { 
+                BOOST_CHECK(element.mightVanish()); // might disappear during the next call to adapt().
+            }
+            else {
+                BOOST_CHECK(element.isLeaf());
+            }
```

Comment:
Do we want the else? Shouldn't all elements on the leaf grid return true for isLeaf()?